pkg/controller/acmeorders/sync.go

```diff
@@ -163,8 +172,30 @@ func (c *controller) Sync(ctx context.Context, o *cmacme.Order) (err error) {
 			}
 		}
 		return err
+
 	// anyChallengesFailed(challenges) == false is already implied by the above
-	// case, but explicitly check it here in case anything changes in future.
+	// case, but explicitly check it in the following cases for if anything changes in future.
+
+	// This is to avoid stuck Orders in edge cases where all the Challenges have
+	// been finalized, but the ACME server has not yet updated the ACME Order's
+	// status to valid. This is not an expected behaviour from an ACME server
+	// https://tools.ietf.org/html/rfc8555#section-7.1.6
+	// https://github.com/jetstack/cert-manager/issues/2868
+	case !anyChallengesFailed(challenges) && allChallengesFinal(challenges) && acmeOrder.Status == acmeapi.StatusPending:
+		log.V(logf.InfoLevel).Info("All challenges in a final state, waiting for ACME server to update the status of the order...")
+		// This is probably not needed as at this point the Order's status
+		// should already be Pending, but set it anyway to be explicit.
+		c.setOrderState(&o.Status, string(cmacme.Pending))
+		// Re-queue the Order to be processed after RequeuePeriod
```

Comment:
This comment is in the wrong place and it seems that the RequeuePeriod global variable is not used